### ISSUE TITLE
SC2034 - False positive when using -v to check with variable

### ISSUE THREAD

**Issue by bgardner-noggin** [koalaman/shellcheck#2461]
#### For bugs
- SC2034
- Shellcheck 0.72
- [X] The rule's wiki page does not already cover this (partially covered, see below)
- [-] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit (cannot recreate on shellcheck.net, but changelog for 0.8.0 does not mention SC2034)

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

declare -A FOO

FOO[bar]=1      # <---- Warning here that FOO appears unused

declare -a KEYS
KEYS=(abc bar)

#Uncommenting this and warning above disappears
#if [[ -v "FOO[bar]" ]]; then
#	echo "Referenced"
#fi

for KEY in "${KEYS[@]}"; do
	if [[ -v "FOO[$KEY]" ]]; then
		echo "$KEY Exists in \$FOO"
	else
		echo "$KEY Not exists in \$FOO"
	fi

	#Uncommenting this and warning above disappears
	#echo "${FOO[$KEY]}"
done


```

#### Here's what shellcheck currently says:
SC2034

#### Here's what I wanted or expected to see:
No SC2034 warning

#### Re wiki page for SC2034
SC2034 mentions that it will not follow indirect references, but note in my example above that

```sh
echo ${FOO[$KEY]}
```

Causes the warning to disappear, but

```sh
if [[ -v "FOO[$KEY]" ]];
```sh

does not, ie one type of indirect reference does work, but the other does not.

**Comment by pikrzysztof** [comment]
A simpler test case for the same bug I believe:

```sh
#!/bin/bash

x="something"
if test -v x; then
        echo "good"
else
        echo "bad"
fi
```

#### Here's what shellcheck currently says:
```
Line 3	SC2034: x appears unused. Verify use (or export if used externally).
```
#### Here's what I wanted or expected to see:

no errrors, `x` is used - we test if it's set.
